src/mscorlib/src/System/TimeZoneInfo.cs

```diff
@@ -1934,9 +1934,36 @@ private class OffsetAndRule {
 
 
 #if PLATFORM_UNIX
-        static public TimeZoneInfo FindSystemTimeZoneById(string id) {
-            // UNIXTODO
-            throw new NotImplementedException();
+        static public TimeZoneInfo FindSystemTimeZoneById(string id)
+        {
+            if (string.IsNullOrEmpty(id))
+            {
+                throw new ArgumentNullException("id");
+            }
+
+            // Special case for Utc since there is code that expects only 1 instance of the UTC time zone info to be created.
+            // see CachedData.GetCorrespondingKind()
+            if (String.Compare(id, c_utcId, StringComparison.OrdinalIgnoreCase) == 0)
+            {
+                return TimeZoneInfo.Utc;
+            }
+
+            // UNIXTODO: use CachedData
+
+            string timeFilePath = Path.Combine(c_timeZoneDirectory, id);
+            if (!File.Exists(timeFilePath))
+            {
+                throw new TimeZoneNotFoundException(Environment.GetResourceString("TimeZoneNotFound_MissingRegistryData", id));
+            }
+
+            byte[] rawData = File.ReadAllBytes(timeFilePath);
```

Comment:
Well, if it'll be common that the file doesn't exist, there's a perf benefit to avoiding an exception if possible.